drools-wb-screens/drools-wb-guided-scorecard-editor/drools-wb-guided-scorecard-editor-client/src/main/java/org/drools/workbench/screens/guided/scorecard/client/handlers/NewGuidedScoreCardHandler.java

```diff
@@ -65,19 +83,24 @@ public ResourceTypeDefinition getResourceType() {
     }
 
     @Override
-    public void create( final Package pkg,
-                        final String baseFileName,
-                        final NewResourcePresenter presenter ) {
-        final ScoreCardModel model = new ScoreCardModel();
-        model.setName( baseFileName );
-        model.setPackageName( pkg.getPackageName() );
-        busyIndicatorView.showBusyIndicator( CommonConstants.INSTANCE.Saving() );
-        scoreCardService.call( getSuccessCallback( presenter ),
-                               new HasBusyIndicatorDefaultErrorCallback( busyIndicatorView ) ).create( pkg.getPackageMainResourcesPath(),
-                                                                                                       buildFileName( baseFileName,
-                                                                                                                      resourceType ),
-                                                                                                       model,
-                                                                                                       "" );
+    public boolean canCreate() {
+        return authorizationManager.authorize(PERMISSION,
+                                              sessionInfo.getIdentity());
     }
```

Comment:
I recommend using "authorizationManager.authorize(new ResourceRef("GuidedScoreCardEditor",                                               , ActivityResourceType.EDITOR, ResourceAction.READ, sessionInfo.getIdentity())" instead. This way we avoid having to figure out and build  the permission string "editor.read.?" which is a bit low level.

Reply:
Thanks @dgutierr I've updated the three affected places.